django_etebase/authentication.py

```diff
@@ -0,0 +1,42 @@
+from django.contrib.auth import get_user_model
+from django.utils import timezone
+from fastapi import Depends
+from fastapi.security import APIKeyHeader
+
+from django_etebase.exceptions import AuthenticationFailed
+from django_etebase.token_auth.models import AuthToken, get_default_expiry
+
+User = get_user_model()
+token_scheme = APIKeyHeader(name="Authorization")
+AUTO_REFRESH = True
+MIN_REFRESH_INTERVAL = 60
+
+
+def __renew_token(auth_token: AuthToken):
+    current_expiry = auth_token.expiry
+    new_expiry = get_default_expiry()
+    # Throttle refreshing of token to avoid db writes
+    delta = (new_expiry - current_expiry).total_seconds()
+    if delta > MIN_REFRESH_INTERVAL:
+        auth_token.expiry = new_expiry
+        auth_token.save(update_fields=("expiry",))
+
+
+def get_authenticated_user(api_token: str = Depends(token_scheme)):
+    api_token = api_token.split()[1]
+    try:
+        token: AuthToken = AuthToken.objects.select_related("user").get(key=api_token)
```

Comment:
I'm not sure whether this correct or not. I assume this whole function is called in an async context (even if this function itself is not async), so I think this line needs to be converted to async `sync_to_async` as mentioned in https://docs.djangoproject.com/en/3.1/topics/async/ and this whole function probably needs to be async.

What do you think?

Reply:
all db calls should be wrapped , its working async now , looks kinda clunky maybe later it can be refactored to something nicer

Reply:
How you did it looks good!